third-party/pjproject/patches/0001-Add-full-support-for-SHA-256-and-SHA-512-256-digest-.patch

```diff
@@ -0,0 +1,2443 @@
+-                                          method);
+-            } else {
+-                status = pjsip_auth_create_digest( &cred->response, 
+-                                          &cred->nonce, &cred->nc, 
+-                                          &cred->cnonce, &pjsip_AUTH_STR, 
+-                                          uri, &chal->realm, 
+-                                          cred_info, method);
+-            }
++            status = pjsip_auth_create_digest2(
++                                      &cred->response, &cred->nonce,
++                                      &cred->nc, &cred->cnonce,
++                                      &pjsip_AUTH_STR, uri,
++                                      &chal->realm, cred_info,
++                                      method, challenge_algorithm_type);
+         }
+ 
+     } else {
+         /* Server requires quality protection that we don't support. */
+         PJ_LOG(4,(THIS_FILE, "Unsupported qop offer %.*s",
+                   (int)chal->qop.slen, chal->qop.ptr));
+         return PJSIP_EINVALIDQOP;
+     }
+ 
+     return status;
+@@ -682,43 +699,72 @@ static void update_digest_session( pjsip_cached_auth *cached_auth,
+             }
+             cached_auth->nc = 1;
+         }
+     }
+ }
+ #endif  /* PJSIP_AUTH_QOP_SUPPORT */
+ 
+ 
+ /* Find cached authentication in the list for the specified realm. */
+ static pjsip_cached_auth *find_cached_auth( pjsip_auth_clt_sess *sess,
+-                                            const pj_str_t *realm )
++                                            const pj_str_t *realm,
++                                            pjsip_auth_algorithm_type algorithm_type)
+ {
+     pjsip_cached_auth *auth = sess->cached_auth.next;
+     while (auth != &sess->cached_auth) {
+-        if (pj_stricmp(&auth->realm, realm) == 0)
++        if (pj_stricmp(&auth->realm, realm) == 0
++            && auth->challenge_algorithm_type == algorithm_type)
+             return auth;
+         auth = auth->next;
+     }
+ 
+     return NULL;
+ }
+ 
+ /* Find credential to use for the specified realm and auth scheme. */
+ static const pjsip_cred_info* auth_find_cred( const pjsip_auth_clt_sess *sess,
+                                               const pj_str_t *realm,
+-                                              const pj_str_t *auth_scheme)
++                                              const pj_str_t *auth_scheme,
++                                              const pjsip_auth_algorithm_type algorithm_type)
+ {
+     unsigned i;
+     int wildcard = -1;
+ 
+     PJ_UNUSED_ARG(auth_scheme);
+ 
+     for (i=0; i<sess->cred_cnt; ++i) {
++        switch(sess->cred_info[i].data_type) {
++        case PJSIP_CRED_DATA_PLAIN_PASSWD:
++            /* PLAIN_PASSWD creds can be used for any algorithm other than AKA */
++            if (algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV1_MD5
++                    || algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV2_MD5) {
```

Comment:
Small change in this commit: https://github.com/pjsip/pjproject/commit/c28635a2d3d27317c9965f7b9182063f20ca5a28.patch
From:
if (algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV1_MD5
+                    || algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV2_MD5) {
To:
            if (algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV1_MD5
                    && algorithm_type != PJSIP_AUTH_ALGORITHM_AKAV2_MD5) {

Reply:
@silentindark Thanks for catching this.  Teluu will be releasing pjproject 2.15 shortly so we'll wait until we update bundled-pjproject to 2.15.  That way we'll get the new stuff plus the fix without needing the patch in third-party/pjproject/patches.

Moving back to draft status.

Reply:
Unresolving so the comment stays visible.